src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js

```diff
@@ -0,0 +1,294 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import TreeItem from '../TreeItem'
+
+import { getKey } from '../../../helpers';
+
+jest.mock('../../../helpers', () => ({
+	getKey: jest.fn(),
+}));
+
+jest.mock('../../../../Icons/ChevronRight', () =>  {
+    const ChevronRight = () => <div className="ChevronRight" />;
+    return ChevronRight;
+});
+
+describe('TreeItem', () => {
+    beforeEach(() => jest.resetAllMocks());
+
+    const getWrapper = (defaultProps, props) => mount(<TreeItem {...defaultProps} {...props} />);
+
+    describe('it', () => {
+        const defaultProps = {
```

Comment:
This test is incomplete!
**first you should test `isSelected` prop**
* if `selected === item`
* if `selected !== item`
* if `selected === getKey(item, keyField)`
* if `selected !== getKey(item, keyField)`
* `getKey` toHaveBeenCalledWith(...)

**Icon prop**
On renderIcon what is heppening.

**ButtonOrLink prop
if href is a function or not

**test for hasChildren prop**

**href prop - toHaveBeenCalledTimes & toHaveBeenCalledWith**

Reply:
1. Added tests for `isSelected `prop:
```
- it('should render with outline prop', () => {
- it('should render without outline prop', () => {
- it('getKey toHaveBeenCalledWith(...)', () => {
```

2. On renderIcon prop func the Icon is rendering. I have this prop in default props:
`renderIcon: jest.fn().mockReturnValue(() => <div />),`
If the renderIcon prop is falsy the Icon should render null:
```
it('should render null', () => {
      expect(
        getCurrentWrapper({ renderIcon: null })
          .find('.TreeItem__icon')
          .exists()
      ).toBe(false);
      expect(
        getCurrentWrapper({ renderIcon: undefined })
          .find('.TreeItem__icon')
          .exists()
      ).toBe(false);
    });
```

3. `it('should render as Link element', () => {` <- if href is a function
`it('should render as button element', () => {` <- if href is not a function

4. I have tested hasChildren prop in `it('should render TreeItem__title--final', () => {` assertion

5. I added the assertion for href prop
`it('should render with href attribute', () => {`